static_frame/core/index.py

```diff
@@ -382,6 +383,16 @@ def copy(self: I) -> I:
         '''
         return self.__copy__() #type: ignore
 
+    def __sizeof__(self: I) -> I:
+        # TODO: Could labels and positions be is-equal?
+        return sum(getsizeof(o) for o in (
```

Comment:
It might be a good idea to add a test that validates against the `__slots__` attribute, to guard against someone adding something to the class and forgetting to update the `__sizeof__` method.

My thought would be something like:

```python
def alternative_get_size(index: Index) -> int:
    return sum(getsizeof(getattr(index, attr) for attr in index.__slots__))

assert alternative_get_size(index) == getsizeof(index)
```

Same would probably go for the other containers

Reply:
That's a beautiful test idea!

Reply:
@chaburkland : that will not work, as `__slots__` only defines attributes at *that* level in the object hierarchy: it does not tell you all attributes that are available (slotted) on that class (via its parents' slots).

Reply:
Good point. I think you could use the pattern of defining module-level tuple for the slots (like you do in `index.py::_INDEX_SLOTS`) that `__slots__` is set to. Then you can test against that instead of `obj.__slots__`

Reply:
Could that tuple be generated automatically? I'd rather not add an extra place to update, particularly if the purpose is to catch when people forget to update a different section of code

Reply:
I think I wasn't very clear. My suggestion is this (you can look at `index.py` to see the existing precedent in the code base).

```python
_FRAME_SLOTS = (
            '__weakref__',
            '_blocks',
            '_columns',
            '_index',
            '_name',
            )

class Frame(ContainerOperand):
    __slots__ = _FRAME_SLOTS
```

Now, the test can just look like this:
```python
# test_frame.py
from static_frame.core.frame import _FRAME_SLOTS

def alternative_get_size(frame: Frame) -> int:
    return sum(getsizeof(getattr(index, attr) for attr in _FRAME_SLOTS))

assert alternative_get_size(frame) == getsizeof(frame)
```

Reply:
Added `sizeof_helper` (final name tbd) to automatically collect the slots in each level of the hierarchy. I think the test cases should check that all class attributes are accounted for manually.